metrics/engine/ingester_test.go

```diff
@@ -107,3 +222,43 @@ func newTestPreInitState(tb testing.TB) *lib.TestPreInitState {
 		BuiltinMetrics: metrics.RegisterBuiltinMetrics(reg),
 	}
 }
+
+func cardinalityMap() {
+	m := make(map[uint64]struct{})
+	for i := 1; i <= 100000; i++ {
+		m[uint64(i)] = struct{}{}
```

Comment:
hmm I am not sure this benchmark accurately represents the reality :thinking: it creates a new `uint64` allocation here, but in our case, we would just be reusing the already existing atlas pointer, right? :thinking:

Reply:
I'm using those benchmarks only for the B/op comparison. For ns/op or allocs/op these tests are not representative.

>hmm I am not sure this benchmark accurately represents the reality

In the case this is for B/op, why it shouldn't be realistic? I expect that a key of a pointer has mostly the same size as a uint64, because under the hood it should be a hash of the struct pointer. Am I missing something?